Frontend Overhaul blog post (#1877)
open-telemetry/opentelemetry.io: content/en/blog/2022/frontend-overhaul/index.md

---

> title: Front end Overhaul of the OpenTelemetry Demo (Go to Next.js)

Reviewer: "Frontend", "Front-end" or "Front end"?

https://en.wikipedia.org/wiki/Front-end_web_development

Reviewer 2: I think I saw "back-end" used a lot across our docs, so it would be "Front-end" then

Reviewer 3: Sold!

[OpenTelemetry Community Demo SIG](https://github.com/open-telemetry/opentelemetry-demo).
The SIG supports a set of instrumented microservices and a front end web app
which are used to show how to instrument a distributed system with OpenTelemetry.

The main focus of the web app is to demonstrate how to instrument an application,
no matter what programming language, platform or OS it uses. The web app also
shows different instrumentation techniques: automatic and manual, metrics,
and baggage. All while following the standards and conventions proscribed
in the official OTel documentation. More about the specific requirements can be
[found here](https://github.com/open-telemetry/opentelemetry-demo/tree/main/docs/requirements).

My company was focused on becoming part of and embracing the OpenTelemetry
community. One of our goals this summer was to get more involved with a core
OpenTelemetry project where we could provide a meaningful contribution.
The OTel demo was the best match for our goal, as contributing would not
only help the community, but also provide a great example to test
and showcase our product with.

> The first thing we did was to get in contact with Carter Socha, the organizer of

Reviewer: ```suggestion
The first thing we did was to get in contact with [Carter Socha](link to GitHub profile), the organizer of
```

the OTel Demo SIG. Carter was really welcoming and helped us identify where our
contributions could be the most impactful. We started looking at
[the issue created by Austin Parker](https://github.com/open-telemetry/opentelemetry-demo/issues/39)
referencing a complete front-end overhaul that would involve moving the

> application away from Go (Server Side Render) to an architecture that included a browser-side

Reviewer 3: If you want to use the acronym SSR you totally can. Best practice is to explain the acronym the first time you use it in a post, and then use the acronym after that. 
eg:
" ... away from Go Server-Side Renderer (SSR) to an architecture that included a browser-side client (Client-Side Renderer or CSR), as well as ..."

client (Client Side Render), as well as improving the overall style, theme, and user
experience.

A fun aspect of the work was the request to move the store from a "normal" store
to an astronomy store to match the OpenTelemetry project’s overall branding.

Once we got the green light from the rest of the OTel demo SIG,
then we started working on the different changes that were part
of the front end architecture overhaul.

## OpenTelemetry Demo Application Description and Tech Stack

The demo app is an astronomy store, with basic eCommerce features such as a
shopping cart, currency selector, payment, and checkout. It also includes
the ability to display promotions (ads) and related products depending on
the user’s context.

The demo’s stack includes multiple microservices in different languages,
covering each of the following OTel-supported languages:

1. Go
2. PHP
3. C++
4. .NET
5. Ruby
6. Python
7. Node.js
8. Rust

> 9. Elixir

Reviewer: Alphabetic ordering?

Every microservice has a specific goal and can communicate with others by using

> a global GRPC definition. Persistent information is saved into a Postgres

Reviewer: https://en.wikipedia.org/wiki/GRPC

```suggestion
a global gRPC definition. Persistent information is saved into a Postgres
```

database and there are outbound services that connect with third-party services
to trigger events (such as confirmation emails). All of the microservices,

> including the front end, are connected to the same Open Telemetry collector

Reviewer: ```suggestion
including the front end, are connected to the same OpenTelemetry collector
```

> instance, which uses Jaeger as one of the data stores for the traces and spans.

Reviewer: It sounds like traces and spans are different things, consider some rewording.

![Open Telemetry Demo System Diagram](system-diagram.png)

> Prior to re-architecting, the front end consisted of a Golang Server Side Rendered app,

Reviewer: Why uppercase "Server Side Rendered"?

Reviewer 3: Since it's been explained earlier in the post, the first time the acronym was used (I think it's line 33), you can use "SSR" here onwards.

help us achieve this by creating a modernized version of the application. It
included an improved way to display the products landing page, an updated
product details page, a mini cart, and a fully compatible mobile version of the
application.

![Open Telemetry Demo New Front end](new-design.png)

Now we had an application design that would match the rest of the OpenTelemetry
themes and colors and look more like the OpenTelemetry.io website.

## Front end Application Architecture Overhaul

We worked on an initial proposal that included the following:

- Framework and tooling (Scaffolding, I/O, styling, UI library)
- Code Architecture and structure (Directories, coding patterns)
- Instrumentation
- Deployment & Distribution
- Testing (E2E, unit test)

This proposal was presented to the OpenTelemetry Demo SIG during one of the
weekly Monday meetings and we were given the green light to move ahead. As part
of the changes, we decided to use Next.js to not only work as the primary
front end application but also to work as an aggregation layer between the
front end and the GRPC back end services.

![New Front end Data Flow](data-flow.png)

> As you can see in the Diagram, the application has two major connectivity

Reviewer: ```suggestion
As you can see in the diagram, the application has two major connectivity
```

A simple back end solution was designed using the
[official GRPC exporter](https://www.npmjs.com/package/@opentelemetry/exporter-trace-otlp-grpc)
in combination with the
[Nodejs SDK](https://www.npmjs.com/package/@opentelemetry/sdk-node).

You can find the full
[implementation here](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/frontend/utils/telemetry/Instrumentation.js).
The basic instrumentation includes auto instrumentation for most of the commonly
used
[libraries and tools for Node.js](https://www.npmjs.com/package/@opentelemetry/auto-instrumentations-node).
As part of providing a better example for users, a manual instrumentation in the
form of route middleware was added. This would catch the incoming HTTP
request and create a span based on it, including the context propagation. The
[implementation can be found here](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/frontend/utils/telemetry/InstrumentationMiddleware.ts).

The front end was a little trickier, as the
[initial rendering is server-side](https://nextjs.org/learn/foundations/how-nextjs-works/rendering).
We had to make sure to load the tracer from the browser side when

> the Javascript code is executed.

Reviewer: ```suggestion
the JavaScript code is executed.
```

After adding validations to check the browser side, we then loaded the custom
front end tracing module, which included creating the
[web tracer provider and the automatic web instrumentations](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/frontend/utils/telemetry/FrontendTracer.ts).

The automatic front end instrumentation captures the most common user actions such as clicks,
fetch requests, and page loads. In order to allow the browser side to interact with the collector,
a config change is needed: enable incoming CORS requests from the web app.

> Once the setup is complete, by loading the application from docker and

Reviewer: ```suggestion
Once the setup is complete, by loading the application from Docker and
```

interacting with the different features, we can start looking at the full traces
that begin from the front end user events all the way to the back end GRPC
services.

![Front end Trace Jaeger Visualization](jaeger.png)

## Contributing to OpenTelemetry was Rewarding

As a team focused on building an open source tool in the Observability space,
the opportunity to help the overall OpenTelemetry community was important to us.
In addition, having a complex microservice-based application that uses multiple
different languages and technologies is directly useful for our team. We really
enjoyed the process of making a contribution to the OpenTelemetry project and
are actively looking for more opportunities to contribute!

> _Oscar Reyes and Olly Babiak also are working on Tracetest, an open source tool

Reviewer: Consider link to GitHub profiles.

application.

![OpenTelemetry Demo New Front-end](new-design.png)

Now we had an application design that would match the rest of the OpenTelemetry
themes and colors and look more like the OpenTelemetry.io website.

## Front-end Application Architecture Overhaul

We worked on an initial proposal that included the following:

- Framework and tooling (Scaffolding, I/O, styling, UI library)
- Code Architecture and structure (Directories, coding patterns)
- Instrumentation
- Deployment & Distribution
- Testing (E2E, unit test)

This proposal was presented to the OpenTelemetry demo SIG during one of the
weekly Monday meetings and we were given the green light to move ahead. As part
of the changes, we decided to use [Next.js](https://nextjs.org/) to not only
work as the primary front-end application but also to work as an aggregation
layer between the front-end and the gRPC back-end services.

![New Front-end Data Flow](data-flow.png)

As you can see in the diagram, the application has two major connectivity
points, one coming from the browser side (REST) to connect to the Next.js
aggregation layer and the other from the aggregation layer to the back-end
services (gRPC).

> ## OpenTelemetry Instrumentation

Reviewer 4: One thing that would be really great to have here, and I won't block on this but I still think it'd be helpful, would be an elaboration on how instrumentation is plugging into next.js. Specifically, I see that there's a next.js middleware implementation that creates spans. That's cool! But why is it there, and what does it do? How does it get used? I think that would be _very_ interesting to people who are also getting into tracing with frontends.

Reviewer 4: Yep, that sounds neat! Maybe that could also make more sense as a readme or something in the demo repo itself. It's just something I got curious about since the post had me digging through code and it seemed cool but wasn't clear on the approach.

---
title: Front-end Overhaul of the OpenTelemetry Demo (Go to Next.js)

> linkTitle: OTel Demo Front-end Overhaul

Reviewer 5: ```suggestion
linkTitle: Front-end Overhaul Demo
```

> date: 2022-10-18

Reviewer 5: ```suggestion
date: 2022-11-07
draft: true
```

Reviewer 5: Draft status as agreed during Comms meeting.

author: "[Oscar Reyes](https://github.com/xoscar) (Tracetest)"
canonical_url: https://tracetest.io/blog/frontend-overhaul-opentelemetry-demo
---

One of the OpenTelemetry Project's many Special Interest Groups (SIG) is the
[OpenTelemetry Community demo SIG](https://github.com/open-telemetry/opentelemetry-demo).
The SIG supports a set of instrumented microservices and a front-end web app
which are used to show how to instrument a distributed system with
OpenTelemetry.

The main focus of the web app is to demonstrate how to instrument an
application, no matter what programming language, platform or OS it uses. The
web app also shows different instrumentation techniques: automatic and manual,
metrics, and baggage. All while following the standards and conventions

> proscribed in the official OTel documentation. More about the specific

Reviewer 6: typo ```suggestion
prescribed in the official OTel documentation. More about the specific
```

referencing a complete front-end overhaul that would involve moving the
application away from Go server-side render (SSR) to an architecture that
included a browser-side client (client-side render or CSR), as well as improving
the overall style, theme, and user experience.

A fun aspect of the work was the request to move the store from a "normal" store
to an astronomy store to match the OpenTelemetry project’s overall branding.

Once we got the green light from the rest of the OTel demo SIG, then we started
working on the different changes that were part of the front-end architecture
overhaul.

## OpenTelemetry Demo Application Description and Tech Stack

The demo app is an astronomy store, with basic eCommerce features such as a
shopping cart, currency selector, payment, and checkout. It also includes the
ability to display promotions (ads) and related products depending on the user’s
context.

The demo’s stack includes multiple microservices in different languages,
covering each of the following OTel-supported languages:

- .NET
- C++
- Elixir
- Go
- Node.js
- PHP
- Python
- Ruby

> - Rust

Reviewer 5: Would it make sense to make these links into the corresponding language SIG's landing page in the website?

Author: Hello @chalin do you know where I can find the links to the SIG's landing page?

Reviewer 5: I was referring to the listing of languages under https://opentelemetry.io/docs/instrumentation/

Author: Done 😄

> - [.NET](https://opentelemetry.io/docs/instrumentation/net/)
> - [C++](https://opentelemetry.io/docs/instrumentation/cpp/)
> - [Erlang/Elixir](https://opentelemetry.io/docs/instrumentation/erlang/)
> - [Go](https://opentelemetry.io/docs/instrumentation/go/)
> - [Node.js](https://opentelemetry.io/docs/instrumentation/js/)
> - [PHP](https://opentelemetry.io/docs/instrumentation/php/)
> - [Python](https://opentelemetry.io/docs/instrumentation/python/)
> - [Ruby](https://opentelemetry.io/docs/instrumentation/ruby/)
> - [Rust](https://opentelemetry.io/docs/instrumentation/rust/)

Reviewer 5: Changes:

- Use local paths not external URLs here
- Alphabetical order of languge names: put .NET after C++ (as if it were named dotnet).

```suggestion
- [C++](/docs/instrumentation/cpp/)
- [.NET](/docs/instrumentation/net/)
- [Erlang/Elixir](/docs/instrumentation/erlang/)
- [Go](/docs/instrumentation/go/)
- [Node.js](/docs/instrumentation/js/)
- [PHP](/docs/instrumentation/php/)
- [Python](/docs/instrumentation/python/)
- [Ruby](/docs/instrumentation/ruby/)
- [Rust](/docs/instrumentation/rust/)
```

- Code Architecture and structure (Directories, coding patterns)
- Instrumentation
- Deployment & Distribution
- Testing (E2E, unit test)

This proposal was presented to the OpenTelemetry demo SIG during one of the
weekly Monday meetings and we were given the green light to move ahead. As part
of the changes, we decided to use [Next.js](https://nextjs.org/) to not only
work as the primary front-end application but also to work as an aggregation
layer between the front-end and the gRPC back-end services.

![New Front-end Data Flow](data-flow.png)

As you can see in the diagram, the application has two major connectivity
points, one coming from the browser side (REST) to connect to the Next.js
aggregation layer and the other from the aggregation layer to the back-end
services (gRPC).

## OpenTelemetry Instrumentation

The next big thing we worked was a way to instrument both sides of the Next.js
app. To do this we had to connect the app twice to the same collector used by
all the microservices.

A simple back-end solution was designed using the
[official gRPC exporter](https://www.npmjs.com/package/@opentelemetry/exporter-trace-otlp-grpc)
in combination with the

> [Nodejs SDK](https://www.npmjs.com/package/@opentelemetry/sdk-node).

Reviewer 5: Nit:
```suggestion
[Node.js SDK](https://www.npmjs.com/package/@opentelemetry/sdk-node).
```

---
title: Frontend Overhaul of the OpenTelemetry Demo (Go to Next.js)

> linkTitle: Frontend Overhaul

Reviewer 2: ```suggestion
linkTitle: OTel Demo Frontend Overhaul
```

date: 2022-10-18
author: "[Oscar Reyes](https://github.com/xoscar) (Tracetest)"

> draft: true

Reviewer 2: remove

canonical_url: https://tracetest.io/blog/frontend-overhaul-opentelemetry-demo
---

One of the OpenTelemetry Project's many Special Interest Groups (SIG) is the

> OpenTelemetry Community Demo SIG which gives support to a set of instrumented

Reviewer 2: ```suggestion
[OpenTelemetry Community Demo SIG](https://github.com/open-telemetry/opentelemetry-demo) which gives support to a set of instrumented
```

techniques (automatic and manual instrumentation, metrics, baggage). All of this
while following the standards and conventions defined by the official
OpenTelemetry Documentation. More about the specific requirements can be
[found here](https://github.com/open-telemetry/opentelemetry-demo/tree/main/docs/requirements).

My company was focused on becoming part of and embracing the OpenTelemetry
community. One of our goals this summer was to get more involved with a core
OpenTelemetry project where we could provide a meaningful contribution. The OTel
demo became the best match for achieving that goal as it would not only help the
community, but help our product as we needed a good example to test and showcase
what can be done with our tool.

The first thing we did was to get in contact with Carter Socha, the organizer of
the OTel Demo SIG. Carter was really welcoming and helped us identify where our
contributions could be the most impactful. We started looking at
[the issue created by Austin Parker](https://github.com/open-telemetry/opentelemetry-demo/issues/39)
referencing a complete front-end overhaul that would involve moving the

> application away from Go (SSR) to an architecture that included a browser-side

Reviewer 2: what does SSR, CSR stand for? don't expect people to know that, introduce those abbreviation or link to a definition eg wikipedia

client (CSR), as well as improving the overall style, theme, and user
experience.

> A fun aspect of the work was the request to move the store from a ‘normal’ store

Reviewer 2: ```suggestion
A fun aspect of the work was the request to move the store from a "normal" store
```

the application frontend architecture overhaul.

## OpenTelemetry Demo Application Description and Tech Stack

The demo app is an astronomy webstore that has the basic functionality to
purchase online products such as a shopping cart, currency selector, product
listing, and payment & checkout. It also includes features to display
promotional items (ads) and related products depending on the context of the
user. The demo stack includes a set of multiple microservices built in different
languages, one for each of the following programming languages:

1. Go
2. PHP
3. C++
4. .NET
5. Ruby
6. Python
7. Node.js
8. Rust
9. Elixir

Every microservice has a specific goal and can communicate with others by using
a global GRPC definition. Persistent information is saved into a Postgres
database and there are outbound services that connect with third-party services
to trigger events (such as confirmation emails). All of the microservices,
including the front end, are connected to the same Open Telemetry collector
instance, which uses Jaeger as one of the data stores for the traces and spans.

![Open Telemetry Demo System Diagram](system-diagram.png)

> The front end was constituted by a Golang SSR application which sent the

Reviewer 3: Rewording:
“Prior to re-architecting, the front end consisted of a Golang SSR app, which sent complete HTML to the browser for display. Every request and call redirected to the server so new information was shown.”

was redirecting the user back to the server so the new piece of information was
shown.

## Web App Styling Improvements, Theme Updates, and User Experience Redesign

Before starting the development process, the frontend application wasn’t
matching the theme that OpenTelemetry had been using in terms of colors,
products, and overall user experience. In addition, the demo lacked a real
frontend (browser side) application as the current implementation was a Go SSR
application.

![Open Telemetry Demo Old Frontend](old-design.png)

The first task at hand was to bring the demo to the modern age by updating the
design, color schemes, and user experience. Olly Babiak walked into the fray to
help us achieve this by creating a modernized version of the application. It
included an improved way to display the products landing page, an updated
product details page, a mini cart, and a fully compatible mobile version of the
application.

![Open Telemetry Demo New Frontend](new-design.png)

Now we had an application design that would match the rest of the OpenTelemetry
themes and colors and look more like the OpenTelemetry.io website.

## Frontend Application Architecture Overhaul

> We worked on an initial proposal that included the following bullet points:

Reviewer 3: Rewording:
“… that included the following:”

- Framework and tooling (Scaffolding, I/O, styling, UI library)
- Code Architecture and structure (Directories, coding patterns)
- Instrumentation
- Deployment & Distribution
- Testing (E2E, unit test)

> This proposal was presented to the OpenTelemetry Demo SIG duringone of the

Reviewer 3: “duringone” should be “during one”

weekly Monday meetings and we were given the green light to move ahead. As part
of the changes, we decided to use Next.js to not only work as the primary
frontend application but also to work as an aggregation layer between the

> frontend and the GRPC backend services.

Reviewer 3: Keep “front end” or “frontend” consistent throughout. Same with “backend”

![New Frontend Data Flow](data-flow.png)

As you can see in the Diagram, the application has two major connectivity
points, one coming from the browser side (REST) to connect to the Next.js
aggregation layer and the other from the aggregation layer to the backend
services (GRPC).

## OpenTelemetry Instrumentation

> The next big thing we worked on was having a way to instrument both sides of the

Reviewer 3: Rewording:
“… was a way to instrument both sides of the Next.js app. To do this we had to connect the app twice to the same collector used by all the microservices.”

Next.js application. To accommodate this we had to connect the same application
twice to the same open telemetry collector that was being used by all of the
other microservices.

> For the backend side, a simple solution was designed that involved using the

Reviewer 3: Rewording:
“A simple backend solution was designed using the …”

[official GRPC exporter](https://www.npmjs.com/package/@opentelemetry/exporter-trace-otlp-grpc)
in combination with the
[Nodejs SDK](https://www.npmjs.com/package/@opentelemetry/sdk-node).

You can find the full
[implementation here](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/frontend/utils/telemetry/Instrumentation.js).
The basic instrumentation includes auto instrumentation for most of the commonly
used
[libraries and tools for Node.js](https://www.npmjs.com/package/@opentelemetry/auto-instrumentations-node).

> But as part of having a better example for users, a manual instrumentation piece

Reviewer 3: Rewording:
“As part of providing a better example for users, a manual instrumentation in the form of route middleware was added. This would catch the …”

was added in the form of a route middleware that would catch the incoming HTTP
request and create a span based on it, including the context propagation. The
[implementation can be found here](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/frontend/utils/telemetry/InstrumentationMiddleware.ts).

> For the frontend side, it became a little bit tricky, as the

Reviewer 3: Rewording:
“The front end was a little trickier, as the rendering is server-side. We had to make sure to …”

[initial Next.js render](https://nextjs.org/learn/foundations/how-nextjs-works/rendering)
comes from the server side so, in this case, we had to make sure to load the
tracer from the browser side when the Javascript code is executed.

> After adding validations to check for the browser side, we then can proceed to

Reviewer 3: Rewording:
“…check the browser side, we then loaded the custom front end tracing module, which included creating the web tracer …”

load the custom front-end tracing module which includes the creation of the
[web tracer provider and the automatic web instrumentations](https://github.com/open-telemetry/opentelemetry-demo/blob/main/src/frontend/utils/telemetry/FrontendTracer.ts).

> The automatic web instrumentations capture the most common user interactions

Reviewer 3: Rewording:
“The automatic front end instrumentation captures the most common user actions such as clicks, fetch requests, and page loads. In order to allow the browser side to interact with the collector, a config change is needed: enable incoming CORS requests from the web app. The resulting collector receivers looks something like this:”

such as click events, fetch requests, and page loads. Another takeaway is that
in order to allow the browser side to interact with the OTel collector there is
a configuration change that needs to be done to enable incoming CORS requests
from the web app. Having the collector receivers configuration looking similar
to this:

> ![OpenTelemetry Collector CORS Config](collector-cors-config.png)

Reviewer 2: don't use a picture, where no picture is needed, some people can not see and some people might want to copy out the content of that picture easily.

Once we got the green light from the rest of the OTel demo SIG, then we started
working on the different changes that were part of the front-end architecture
overhaul.

## OpenTelemetry Demo Application Description and Tech Stack

The demo app is an astronomy store, with basic eCommerce features such as a
shopping cart, currency selector, payment, and checkout. It also includes the
ability to display promotions (ads) and related products depending on the user’s
context.

The demo’s stack includes multiple microservices in different languages,
covering each of the following OTel-supported languages:

- .NET
- C++
- Elixir
- Go
- Node.js
- PHP
- Python
- Ruby
- Rust

Every microservice has a specific goal and can communicate with others by using
a global gRPC definition. Persistent information is saved into a PostgresSQL
database and there are outbound services that connect with third-party services
to trigger events (such as confirmation emails). All of the microservices,
including the front-end, are connected to the same OpenTelemetry collector
instance, which uses Jaeger as one of the data stores for the tracing data.

> ```mermaid

Reviewer 2: while github has a mermaid support, we do not have it in hugo, so you still need to put the picture here.

Author: Ups, sorry about that, I just fixed it with my latest commit. Thanks for the heads up! 😄

metrics, and baggage. All while following the standards and conventions
proscribed in the official OTel documentation. More about the specific
requirements can be
[found here](https://github.com/open-telemetry/opentelemetry-demo/tree/main/docs/requirements).

My company was focused on becoming part of and embracing the OpenTelemetry
community. One of our goals this summer was to get more involved with a core
OpenTelemetry project where we could provide a meaningful contribution. The OTel
demo was the best match for our goal, as contributing would not only help the
community, but also provide a great example to test and showcase our product
with.

The first thing we did was to get in contact with
[Carter Socha](https://github.com/cartersocha), the organizer of the OTel Demo
SIG. Carter was really welcoming and helped us identify where our contributions
could be the most impactful. We started looking at
[the issue created by Austin Parker](https://github.com/open-telemetry/opentelemetry-demo/issues/39)
referencing a complete front-end overhaul that would involve moving the
application away from Go server-side render (SSR) to an architecture that

> included a browser-side client (client-side render), as well as improving the

Reviewer 3: Should probably add the acronym for client-side render, ie:  "...client (client-side render or CSR), as ..."

a global gRPC definition. Persistent information is saved into a PostgresSQL
database and there are outbound services that connect with third-party services
to trigger events (such as confirmation emails). All of the microservices,
including the front-end, are connected to the same OpenTelemetry collector
instance, which uses Jaeger as one of the data stores for the tracing data.

![OpenTelemetry Demo System Diagram](system-diagram.png)

Prior to re-architecting, the front-end consisted of a Golang SSR app, which
sent complete HTML to the browser for display. Every request and call redirected
to the server so new information was shown.

## Web App Styling Improvements, Theme Updates, and User Experience Redesign

Before starting the development process, the front-end application wasn’t
matching the theme that OpenTelemetry had been using in terms of colors,
products, and overall user experience. In addition, the demo lacked a real
front-end (browser side) application as the current implementation was a Go
application.

![OpenTelemetry Demo Old Front-end](old-design.png)

The first task at hand was to bring the demo to the modern age by updating the

> design, color schemes, and user experience. Olly Babiak walked into the fray to

Reviewer 3: Would be good to provide a link to Olly Babiak's GitHub profile (or Twitter if GH not available).

> This proposal was presented to the OpenTelemetry Demo SIG during one of the

Reviewer 3: The SIG has been referred to as "demo SIG" throughout, suggest applying the lowercase 'd' here for consistency. Not a blocking item!
